theano/sparse/tests/test_basic.py

```diff
@@ -464,6 +464,12 @@ def test_dot(self):
                                config.floatX, 3))],
                 Dot)
 
+    def test_dot_broadcast(self):
+        A = sp.matrix('csr')
+        b = tensor.vector()
+        bc = sp.dot(A, b[:, None]).broadcastable
+        assert bc == (False, True)
```

Comment:
Can you test with something like:
```
for x, y in [
(SparseType('csr', float32), tensor.vector()),
(SparseType('csr', float32), tensor.vector()[:, None]),
(SparseType('csr', float32), tensor.vector()[None, :]),
(SparseType('csr', float32), tensor.matrix()),
(tensor.vector(), SparseType('csr', float32), 
(tensor.vector()[:, None], SparseType('csr', float32), 
(tensor.vector()[None, :], SparseType('csr', float32), 
(tensor.matrix(), SparseType('csr', float32)]

sparse_out = sp.dot(x, y)
if isinstance(x, SparseVariable):
    x = tensor.matrix()
if isinstance(y, SparseVariable):
    y = tensor.matrix()
dense_out = tensor.dot(x, y)
assert dense_out.broadcastable == sparse_out.broadcastable
```

This would fix travis (the current line 468 fail) and would make sure we test all cases.

thanks.